src/utils/helpers.ts

```diff
@@ -84,12 +84,14 @@ export function findNestedNode(
   }
 
   // Check for nested children
-  const nodesWithChildren = children.filter(n => n.children);
+  const nodesWithChildren = children.filter(
+    n => n.children && n.children.length > 0
```

Comment:
Could be reduced to:

```suggestion
    n => n.children?.length
```